dynawo/sources/Modeler/DataInterface/DYNCriteria.cpp

```diff
@@ -0,0 +1,213 @@
+      return false;
+    }
+  }
+  return true;
+}
+
+bool
+BusCriteria::criteriaEligibleForBus(const boost::shared_ptr<criteria::CriteriaParams>& params) {
+  if (params->getType() == criteria::CriteriaParams::SUM) {
+    Trace::warn() << DYNLog(SumBusCriteriaIgnored) << Trace::endline;
+    return false;
+  }
+  if (!params->hasUMinPu() && !params->hasUMaxPu())
+    return false;
+  if (params->hasPMax())
+    Trace::warn() << DYNLog(PowerBusCriteriaIgnored) << Trace::endline;
+  if (params->hasPMin())
+    Trace::warn() << DYNLog(PowerBusCriteriaIgnored) << Trace::endline;
+  return true;
+}
+
+void
+BusCriteria::addBus(const boost::shared_ptr<BusInterface>& bus) {
+  if (params_->hasUMinNom() &&
+      bus->getVNom() < params_->getUMinNom()) return;
+  if (params_->hasUMaxNom() &&
+      bus->getVNom() > params_->getUMaxNom()) return;
+  if (doubleEquals(bus->getV0(), defaultV0) || bus->getV0() <  defaultV0) return;
+  buses_.push_back(bus);
+}
+
+LoadCriteria::LoadCriteria(const boost::shared_ptr<criteria::CriteriaParams>& params) :
+                Criteria(params) {}
+
+LoadCriteria::~LoadCriteria() {}
+
+bool
+LoadCriteria::checkCriteria(bool finalStep) const {
+  if (!finalStep && params_->getScope() == criteria::CriteriaParams::FINAL)
+    return true;
+  double sum = 0.;
+  bool atLeastOneEligibleLoadWasFound = false;
+  for (std::vector<boost::shared_ptr<LoadInterface> >::const_iterator it = loads_.begin(), itEnd = loads_.end();
+      it != itEnd; ++it) {
+    double p = (*it)->getP();
+    if ((params_->hasUMaxPu() || params_->hasUMinPu()) && (*it)->getBusInterface()) {
+      double v = (*it)->getBusInterface()->getStateVarV();
+      double vNom = (*it)->getBusInterface()->getVNom();
+      if (params_->hasUMaxPu() && v > params_->getUMaxPu()*vNom) continue;
+      if (params_->hasUMinPu() && v < params_->getUMinPu()*vNom) continue;
+    }
+    if (params_->getType() == criteria::CriteriaParams::LOCAL_VALUE) {
+      if (params_->hasPMax() && p > params_->getPMax()) {
+        Trace::debug() << DYNLog(SourceAbovePower, (*it)->getID(), p, params_->getPMax(), params_->getId()) << Trace::endline;
+        return false;
+      }
+      if (params_->hasPMin() && p < params_->getPMin()) {
+        Trace::debug() << DYNLog(SourceUnderPower, (*it)->getID(), p, params_->getPMin(), params_->getId()) << Trace::endline;
+        return false;
+      }
+    } else {
+      sum+=p;
+      atLeastOneEligibleLoadWasFound = true;
+    }
+  }
+
+  if (atLeastOneEligibleLoadWasFound && params_->getType() == criteria::CriteriaParams::SUM) {
+    if (params_->hasPMax() && sum > params_->getPMax()) {
+      Trace::debug() << DYNLog(SourcePowerAboveMax, sum, params_->getPMax(), params_->getId()) << Trace::endline;
+      return false;
+    }
+    if (params_->hasPMin() && sum < params_->getPMin()) {
+      Trace::debug() << DYNLog(SourcePowerBelowMin, sum, params_->getPMin(), params_->getId()) << Trace::endline;
```

Comment:
I have difficulties imagining a plausible criterion with this form with the current possibilities. Do you have in mind some possible examples? What is your opinion on this?

Reply:
Yes, I agree this is a corner case, but I don't see a good reason to disable this possibility for generality reasons

Reply:
I was also hesitating what to do with this. Thus it is fine.